source/extensions/filters/network/redis_proxy/config.cc

```diff
@@ -14,6 +18,15 @@ namespace Extensions {
 namespace NetworkFilters {
 namespace RedisProxy {
 
+// This is a slightly unique way of registering for the singleton manager, as the Redis redirection
```

Comment:
A couple of things here:
1) I would put the redirection manager inside source/extensions/common/redis since it's shared by both the filter and cluster code.
2) Technically this string code here violates the init fiasco rule. In order to handle this, my advice is to actually have a helper function inside the common code that gets the singeton, and hides all of that behind a free function. I did something similar here which you can model it from: https://github.com/envoyproxy/envoy/blob/cc057ce626adb2076d3dddb8dec4628c62fc5b35/source/extensions/common/dynamic_forward_proxy/dns_cache.h#L168-L173

Reply:
I've just pushed up code changes to address these issues. Hopefully, it will pass all of the CI checks.